src/sparseml/pytorch/sparsification/distillation/modifier_per_layer.py

```diff
@@ -328,24 +368,51 @@ def compute_distillation_loss(self, optimizer: torch.optim.Optimizer, **kwargs):
         if self.project_features and len(self._projection) == 0:
             # NOTE: have to call initialize here because we need the cached output
             # from the module. i.e. we need forward to have been called already
-            projection_params = self._initialize_projection()
-            optimizer.add_param_group({"params": projection_params})
-
-            # NOTE: since we added a param group, if someone tries
-            # to serialize the optimizer, the new param group will be
-            # included. if they try call load_state_dict() on
-            # a fresh optimizer (one that hasn't been called
-            # with compute_distillation_loss), it will
-            # crash.
-            _old_state_dict_fn = optimizer.state_dict
-
-            def state_dict_without_latest_param_group():
-                full_state_dict = _old_state_dict_fn()
-                original_state_dict = deepcopy(full_state_dict)
-                original_state_dict["param_groups"].pop()
-                return original_state_dict
+            self._projection = self._initialize_projection()
+
+            if self._loaded_projection is not None:
+                for name, layer in self._projection.items():
+                    layer.load_state_dict(self._loaded_projection.pop(name))
+                self._loaded_projection = None
+
+        if _get_projection_param_group_idx(optimizer.param_groups) is None:
+            optimizer.add_param_group(
+                {
+                    _DISTILL_PARAM_GROUP_KEY: True,
+                    "params": [p.weight for p in self._projection.values()],
+                }
+            )
 
-            optimizer.state_dict = state_dict_without_latest_param_group
+            _old_state_dict_fn = optimizer.state_dict
```

Comment:
let's move this block to a helper function

Reply:
Done!